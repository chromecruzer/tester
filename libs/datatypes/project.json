{
  "name": "datatypes",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/datatypes/src",
  "projectType": "library",
  "targets": {
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["libs/datatypes/**/*.ts"]
      }
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/libs/datatypes"],
      "options": {
        "jestConfig": "libs/datatypes/jest.config.ts",
        "passWithNoTests": true
      }
    }
  },
  "tags": []
}
