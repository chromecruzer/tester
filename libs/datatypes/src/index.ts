export * from './datatypes';
export * from './IolRecord';
export * from './CustomerRecord';
export * from './CustomerContactRecord';
export * from './ProductRecord';
export * from './EmployeeRecord';        // added by shyam for testing 
export * from './EmployeeRecordDataMappings'   //
export * from './UserRecord1'  //
export * from './UserRecordDM' // 
export * from './ConsignmentRecord';
export * from './CustomerDataMapping';
export * from './CustomerContactRecord';
export * from './CustomerContactDataMapping';
export * from './IolDataMapping';
export * from './notifications';
export * from './ProductDataMapping';
export * from './SalesRepDataMapping';
export * from './SalesRepRecord';
export * from './SalesMappingSquared';
export * from './SalesMappingRecord';
export * from './SalesJumpRecord';
export * from './UploadHeaders';
export * from './AuditRecord';
export * from './SearchSignature';
export * from './EmailSettings';
//export * from './UserRecord';
import BuildUrl from "./BuildUrl";
import {ExpirationCalculations} from "./ExpirationCalculations";
import ItemMatchStatus from './ItemMatchStatus';
import SearchSignature from "./SearchSignature";
export {BuildUrl, ExpirationCalculations, ItemMatchStatus, SearchSignature};
