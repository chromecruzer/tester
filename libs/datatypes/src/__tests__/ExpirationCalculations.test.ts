import {DateTime} from "luxon";
import {ExpirationCalculations, tracDateFormat} from "@trac/datatypes";
import * as util from "util";
import _ from "lodash";

const dump = (obj, depth = null) => util.inspect(obj, {depth, colors: true});

describe('ExpirationCalculations', () => {
  let twoGroupRecords, expiredRecords, records30, records60, babies, now, calcs;
  beforeEach(() => {
    now = DateTime.fromObject({year: 2021, month: 10, day: 10});
    twoGroupRecords = [
      {
        product_id: 'PREM_IOLS_TOT',
        item: 'premium item 1 border case',
        expire_date: DateTime.fromObject({year:2021, month:12, day:10}).toJSDate(),
      },
      {
        product_id: 'STAND_IOLS_TOT',
        item: 'standard item 1 border case',
        expire_date: DateTime.fromObject({year:2021, month:12, day:20}).toJSDate(),
      },
      {
        product_id: 'PREM_IOLS_TOT',
        item: 'premium item 2',
        expire_date: DateTime.fromObject({year:2021, month:12, day:21}).toJSDate(),
      },
      {
        product_id: 'STAND_IOLS_TOT',
        item: 'standard item 2',
        expire_date: DateTime.fromObject({year:2021, month:12, day:21}).toJSDate(),
      },
      {
        product_id: 'PREM_IOLS_TOT',
        item: 'premium item 3',
        expire_date: DateTime.fromObject({year:2022, month:12, day:20}).toJSDate(),
      },
      {
        product_id: 'STAND_IOLS_TOT',
        item: 'standard item 3',
        expire_date: DateTime.fromObject({year:2022, month:12, day:25}).toJSDate(),
      }
    ];
    expiredRecords = [
      {
        product_id: 'PREM_IOLS_TOT',
        description4: 'premium item 1 border case',
        expire_date: DateTime.fromObject({year:2021, month:10, day:10}).toJSDate(),
      },
      {
        product_id: 'PREM_IOLS_TOT',
        description4: 'premium item 1 border case',
        expire_date: DateTime.fromObject({year:2021, month:10, day:9}).toJSDate(),
      },
      {
        product_id: 'STAND_IOLS_TOT',
        description4: 'standard item 1 border case',
        expire_date: DateTime.fromObject({year:2021, month:4, day:7}).toJSDate(),
      },
    ];
    records30 = [
      {
        product_id: 'PREM_IOLS_TOT',
        description4: 'premium item 1 30 days',
        expire_date: DateTime.fromObject({year:2021, month:11, day:9}).toJSDate(),
      },
      {
        product_id: 'PREM_IOLS_TOT',
        description4: 'premium item 1 30 days',
        expire_date: DateTime.fromObject({year:2021, month:11, day:8}).toJSDate(),
      },
      {
        product_id: 'STAND_IOLS_TOT',
        description4: 'standard item 1 30 days',
        expire_date: DateTime.fromObject({year:2021, month:10, day:31}).toJSDate(),
      },
    ];
    records60 = [
      {
        product_id: 'PREM_IOLS_TOT',
        description4: 'premium item 60 days',
        expire_date: DateTime.fromObject({year:2021, month:11, day:10}).toJSDate(),
      },
      {
        product_id: 'PREM_IOLS_TOT',
        description4: 'premium item 60 days',
        expire_date: DateTime.fromObject({year:2021, month:11, day:11}).toJSDate(),
      },
      {
        product_id: 'STAND_IOLS_TOT',
        description4: 'standard item 60 days',
        expire_date: DateTime.fromObject({year:2021, month:12, day:8}).toJSDate(),
      },
    ];
    babies = [
      {
        product_id: 'PREM_IOLS_TOT',
        description4: 'premium item >60 days',
        expire_date: DateTime.fromObject({year:2022, month:1, day:10}).toJSDate(),
      },
      {
        product_id: 'PREM_IOLS_TOT',
        description4: 'premium item >60 days',
        expire_date: DateTime.fromObject({year:2022, month:1, day:11}).toJSDate(),
      },
      {
        product_id: 'STAND_IOLS_TOT',
        description4: 'standard item >60 days',
        expire_date: DateTime.fromObject({year:2022, month:1, day:8}).toJSDate(),
      },
    ];
    calcs = new ExpirationCalculations();
    calcs.now = now;
  });
  describe('filterRecords', () => {
    it('should only return records with a specific product id', () => {
          const actual = calcs.filterRecords(twoGroupRecords, {standard: true});
          expect(actual).toMatchSnapshot();
    });
  });
  describe('groupConsignments', () => {
    it('should return the correct number of expired lenses', () => {
      const actual = calcs.groupConsignments(expiredRecords, {});
      expect(actual).toMatchSnapshot();
    });
    it('should return the correct number of < 30 lenses', () => {
      const actual = calcs.groupConsignments(records30, {});
      expect(actual).toMatchSnapshot();
    });
    it('should return the correct number of < 60 lenses', () => {
      const actual = calcs.groupConsignments(records60, {});
      expect(actual).toMatchSnapshot();
    });
    it('should return the correct number of > 60 lenses', () => {
      const actual = calcs.groupConsignments(babies, {});
      expect(actual).toMatchSnapshot();
    });
  });
  describe('decorateRecords', () => {
    it('should calculate text date correctly and added the correct label and css', () => {
      const textDateRecords = _.map([...expiredRecords, ...records30, ...records60, ...babies], r => ({...r,
        expire_date: DateTime.fromJSDate(r.expire_date).toFormat(tracDateFormat)}));
      // console.log(`texted dates ${dump(textDateRecords)}`)
      const actual = calcs.decorateRecords(textDateRecords);
      expect(actual).toMatchSnapshot();
    });
  });
});
