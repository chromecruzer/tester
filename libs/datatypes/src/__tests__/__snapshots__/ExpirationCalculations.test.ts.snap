// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpirationCalculations decorateRecords should calculate text date correctly and added the correct label and css 1`] = `
Array [
  Object {
    "description4": "premium item 1 border case",
    "expire_date": "10/10/2021",
    "expire_status": "Expired",
    "expire_status_css": "text-red-600 bg-gray-100 font-extrabold text-center",
    "product_id": "PREM_IOLS_TOT",
  },
  Object {
    "description4": "premium item 1 border case",
    "expire_date": "10/09/2021",
    "expire_status": "Expired",
    "expire_status_css": "text-red-600 bg-gray-100 font-extrabold text-center",
    "product_id": "PREM_IOLS_TOT",
  },
  Object {
    "description4": "standard item 1 border case",
    "expire_date": "04/07/2021",
    "expire_status": "Expired",
    "expire_status_css": "text-red-600 bg-gray-100 font-extrabold text-center",
    "product_id": "STAND_IOLS_TOT",
  },
  Object {
    "description4": "premium item 1 30 days",
    "expire_date": "11/09/2021",
    "expire_status": "Expires In < 60 Days",
    "expire_status_css": "text-bl-text-main bg-gray-100 text-center",
    "product_id": "PREM_IOLS_TOT",
  },
  Object {
    "description4": "premium item 1 30 days",
    "expire_date": "11/08/2021",
    "expire_status": "Expires In < 30 Days",
    "expire_status_css": "text-yellow-300 bg-gray-100 font-bold text-center",
    "product_id": "PREM_IOLS_TOT",
  },
  Object {
    "description4": "standard item 1 30 days",
    "expire_date": "10/31/2021",
    "expire_status": "Expires In < 30 Days",
    "expire_status_css": "text-yellow-300 bg-gray-100 font-bold text-center",
    "product_id": "STAND_IOLS_TOT",
  },
  Object {
    "description4": "premium item 60 days",
    "expire_date": "11/10/2021",
    "expire_status": "Expires In < 60 Days",
    "expire_status_css": "text-bl-text-main bg-gray-100 text-center",
    "product_id": "PREM_IOLS_TOT",
  },
  Object {
    "description4": "premium item 60 days",
    "expire_date": "11/11/2021",
    "expire_status": "Expires In < 60 Days",
    "expire_status_css": "text-bl-text-main bg-gray-100 text-center",
    "product_id": "PREM_IOLS_TOT",
  },
  Object {
    "description4": "standard item 60 days",
    "expire_date": "12/08/2021",
    "expire_status": "Expires In < 60 Days",
    "expire_status_css": "text-bl-text-main bg-gray-100 text-center",
    "product_id": "STAND_IOLS_TOT",
  },
  Object {
    "description4": "premium item >60 days",
    "expire_date": "01/10/2022",
    "expire_status": "",
    "expire_status_css": "text-current",
    "product_id": "PREM_IOLS_TOT",
  },
  Object {
    "description4": "premium item >60 days",
    "expire_date": "01/11/2022",
    "expire_status": "",
    "expire_status_css": "text-current",
    "product_id": "PREM_IOLS_TOT",
  },
  Object {
    "description4": "standard item >60 days",
    "expire_date": "01/08/2022",
    "expire_status": "",
    "expire_status_css": "text-current",
    "product_id": "STAND_IOLS_TOT",
  },
]
`;

exports[`ExpirationCalculations filterRecords should only return records with a specific product id 1`] = `
Array [
  Object {
    "expire_date": 2021-12-20T06:00:00.000Z,
    "item": "standard item 1 border case",
    "product_id": "STAND_IOLS_TOT",
  },
  Object {
    "expire_date": 2021-12-21T06:00:00.000Z,
    "item": "standard item 2",
    "product_id": "STAND_IOLS_TOT",
  },
  Object {
    "expire_date": 2022-12-25T06:00:00.000Z,
    "item": "standard item 3",
    "product_id": "STAND_IOLS_TOT",
  },
]
`;

exports[`ExpirationCalculations groupConsignments should return the correct number of < 30 lenses 1`] = `
Array [
  Object {
    "between30and60": 1,
    "description4": "premium item 1 30 days",
    "lessThan30": 1,
    "moreThan60": 0,
    "tooLate": 0,
    "total": 2,
  },
  Object {
    "between30and60": 0,
    "description4": "standard item 1 30 days",
    "lessThan30": 1,
    "moreThan60": 0,
    "tooLate": 0,
    "total": 1,
  },
]
`;

exports[`ExpirationCalculations groupConsignments should return the correct number of < 60 lenses 1`] = `
Array [
  Object {
    "between30and60": 2,
    "description4": "premium item 60 days",
    "lessThan30": 0,
    "moreThan60": 0,
    "tooLate": 0,
    "total": 2,
  },
  Object {
    "between30and60": 1,
    "description4": "standard item 60 days",
    "lessThan30": 0,
    "moreThan60": 0,
    "tooLate": 0,
    "total": 1,
  },
]
`;

exports[`ExpirationCalculations groupConsignments should return the correct number of > 60 lenses 1`] = `
Array [
  Object {
    "between30and60": 0,
    "description4": "premium item >60 days",
    "lessThan30": 0,
    "moreThan60": 2,
    "tooLate": 0,
    "total": 2,
  },
  Object {
    "between30and60": 0,
    "description4": "standard item >60 days",
    "lessThan30": 0,
    "moreThan60": 1,
    "tooLate": 0,
    "total": 1,
  },
]
`;

exports[`ExpirationCalculations groupConsignments should return the correct number of expired lenses 1`] = `
Array [
  Object {
    "between30and60": 0,
    "description4": "premium item 1 border case",
    "lessThan30": 0,
    "moreThan60": 0,
    "tooLate": 2,
    "total": 2,
  },
  Object {
    "between30and60": 0,
    "description4": "standard item 1 border case",
    "lessThan30": 0,
    "moreThan60": 0,
    "tooLate": 1,
    "total": 1,
  },
]
`;
